std/stdio.d

```diff
@@ -1522,6 +1524,139 @@ for every line.
+
+The file is created with R/W permissions and opened with $(D "w+b"). On POSIX
+systems, the permissions are restricted to the current user, though the
+effective permissions are modified by the process' umask in the usual way.
+
+Params:
+    prefix = Prefix for the generated file name.
+    suffix = Suffix for the generated file name (which also provides a way to
+             give the file an extension).
+    dir    = Directory of the temporary file. Defaults to the result of
+             $(XREF file, tempDir).
+  */
+    static File tempFile(string prefix = null, string suffix = null)
+    {
+        import std.file : tempDir;
+        return File.tempFile(prefix, suffix, std.file.tempDir());
+    }
+
+    /// Ditto
+    static File tempFile(string prefix, string suffix, string dir)
+    {
+        import std.exception : ErrnoException;
+        import std.file : isSymlink;
+        import std.path : absolutePath, baseName, buildPath, dirName;
+
+        static string genTempName(string prefix, string suffix, string dir)
+        {
+            import std.ascii : digits, letters;
+            import std.random : rndGen;
+            import std.range : chain;
+            auto name = new char[](prefix.length + 15 + suffix.length);
+            name[0 .. prefix.length] = prefix;
+            name[$ - suffix.length .. $] = suffix;
+
+            rndGen.popFront();
+            auto chars = chain(cast(immutable(ubyte[]))letters, cast(immutable(ubyte[]))digits);
+            foreach(ref c; name[prefix.length .. $ - suffix.length])
+            {
+                c = chars[rndGen.front % chars.length];
+                rndGen.popFront();
+            }
+
+            return buildPath(dir, name);
+        }
+
+        // Once should be enough given the randomness, but we'll give it three
+        // times to be safe.
+        foreach(i; 0 .. 3)
+        {
+            auto filename = genTempName(prefix, suffix, dir);
+
+            // O_EXCL does not support symbolic links.
+            if(dirName(filename).isSymlink)
+                filename = buildPath(absolutePath(dirName(filename), baseName(filename)));
+
+            import std.internal.cstring : tempCString;
+            version(Posix)
+            {
+                import core.sys.posix.sys.stat, core.sys.posix.fcntl;
+                enum flags = S_IRUSR | S_IWUSR;
+                auto fd = open(tempCString(filename), O_RDWR | O_CREAT | O_EXCL, flags);
+            }
+            else version(Windows)
+            {
+                enum flags = _S_IREAD | _S_IWRITE;
+                auto fd = _wsopen(tempCString!wchar(filename), O_RDWR | O_CREAT | O_EXCL, _SH_DENYNO, flags);
+            }
+            else static assert("Unsupported OS");
+
+            if(fd == -1)
+                continue;
+
+            File retval;
+            retval.fdopen(fd, "w+b", filename);
+            return retval;
+        }
+
+        throw new ErrnoException("Failed to create a temporary file.");
```

Comment:
I don't see why it would be specific to POSIX, and it's used everywhere in std.file. However, that does remind me that I need to indcate that an `ErnoException` can be thrown in `tempFile`'s documentation.